CMakeLists.txt

```diff
@@ -201,16 +203,54 @@ set(CC_SOURCES
 
 add_library(sptk STATIC ${CC_SOURCES})
 target_include_directories(sptk PUBLIC
-  ${PROJECT_SOURCE_DIR}/include
-  ${THIRD_PARTY_DIR}
-  ${THIRD_PARTY_DIR}/REAPER
-  ${THIRD_PARTY_DIR}/WORLD
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}/REAPER>
+  $<BUILD_INTERFACE:${THIRD_PARTY_DIR}/WORLD>
```

Comment:
Do you need to add <INSTALL_INTERFACE>?

Reply:
This is necessary. The include path is only valid, when running cmake from the source dir. When installed, the include path should be something like /usr/local/include. Also, if you leave it out, you will get following errors when installing:

  Target "sptk" INTERFACE_INCLUDE_DIRECTORIES property contains path:

    "/Users/thomas/Developer/sptk/include"

  which is prefixed in the source directory.

Reply:
Sorry, my question is not appropriate. You added `BUILD_INTERFACE` lines but not `INSTALL_INTERFACE`lines.

Do we need to add the following line?
```
$<INSTALL_INTERFACE:include>
```

Reply:
I don't think so. There is this line in the file
`INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}`
This was always enough for me to let cmake do the right thing when installing.

Reply:
> INCLUDES DESTINATION
This option specifies a list of directories which will be added to the [INTERFACE_INCLUDE_DIRECTORIES](https://cmake.org/cmake/help/latest/prop_tgt/INTERFACE_INCLUDE_DIRECTORIES.html#prop_tgt:INTERFACE_INCLUDE_DIRECTORIES) target property of the <targets> when exported by the [install(EXPORT)](https://cmake.org/cmake/help/latest/command/install.html#install-export) command. If a relative path is specified, it is treated as relative to the [$<INSTALL_PREFIX>](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:INSTALL_PREFIX).
source: [https://cmake.org/cmake/help/latest/command/install.html](https://cmake.org/cmake/help/latest/command/install.html)